discovery-infra/delete_nodes.py

```diff
@@ -12,63 +15,113 @@
+    """
+    @utils.on_exception(
+        message='Failed to run terraform delete',
+        silent=True
+    )
+    def _try_to_delete_nodes():
+        log.info('Start running terraform delete')
+        utils.run_command_with_output(
+            f'cd {tf_folder} && '
+            'terraform destroy '
+            '-auto-approve '
+            '-input=false '
+            '-state=terraform.tfstate ' 
+            '-state-out=terraform.tfstate ' 
+            '-var-file=terraform.tfvars.json'
         )
 
+    _try_to_delete_nodes()
+
+    default_network_name = consts.NETWORK_NAME_PREFIX + namespace
+    _delete_virsh_resources(
+        tfvars.get('cluster_name', cluster_name),
+        tfvars.get('libvirt_network_name', default_network_name),
+    )
+
+    log.info('Deleting %s', tf_folder)
+    shutil.rmtree(tf_folder)
+
+
+def _delete_virsh_resources(*filters):
+    log.info('Deleting virsh resources (filters: %s)', filters)
+    virsh_cleanup.clean_virsh_resources(
+        skip_list=virsh_cleanup.DEFAULT_SKIP_LIST,
+        resource_filter=filters
+    )
+
 
-# Deletes every single virsh resource, leaves only defaults
-def delete_all():
-    log.info("Deleting all virsh resources")
-    virsh_cleanup.clean_virsh_resources(virsh_cleanup.DEFAULT_SKIP_LIST, None)
+@utils.on_exception(
+    message='Failed to delete clusters from namespaces',
+    silent=True
+)
+def delete_clusters_from_all_namespaces():
+    for dirname in os.listdir(consts.TF_FOLDER):
+        res = _get_name_and_namespace_from_dirname(dirname)
+        if not res:
+            continue
+        name, namespace = res
+        delete_cluster(name, namespace)
+
+
+def _get_name_and_namespace_from_dirname(dirname):
+    if ':' in dirname:
+        return dirname.rsplit(':', 1)
+
+    log.warning(
+        'Unable to extract cluster name and namespace from directory name %s. '
+        'Directory name convention must be <cluster_name>:<namespace>',
+        dirname
+    )
+
+
+@utils.on_exception(message='Failed to delete nodes', silent=True)
+def delete_cluster(cluster_name, namespace):
+    log.info(
+        'Deleting cluster: %s in namespace: %s',
+        cluster_name, namespace
+    )
+
+    tf_folder = os.path.join(consts.TF_FOLDER, f'{cluster_name}:{namespace}')
+    tfvars = utils.get_tfvars(tf_folder)
+
+    if not args.only_nodes:
+        try_to_delete_cluster(namespace, tfvars)
+    delete_nodes(cluster_name, namespace, tf_folder, tfvars)
 
 
 def main():
     if args.delete_all:
-        delete_all()
-    else:
-        try:
-            tfvars = utils.get_tfvars()
-            if not args.only_nodes:
-                try_to_delete_cluster(tfvars)
-            delete_nodes(tfvars)
-        except:
-            log.exception("Failed to delete nodes")
+        _delete_virsh_resources()
+        return
+
+    if args.namespace == 'all':
+        delete_clusters_from_all_namespaces()
+        return
+
+    cluster_name = args.cluster_name or consts.CLUSTER_PREFIX
```

Comment:
Try to create strings assignment as simple as possible with the right templates, to avoid multiple lines.
```python
cluster_name = f'{args.cluster_name or consts.CLUSTER_PREFIX}-{args.namespace}'
```